flocker/node/agents/gce.py

```diff
@@ -501,32 +555,44 @@ def _get_attached_to(self, blockdevice_id):
-            disk = self._compute.disks().get(project=self._project,
-                                             zone=self._zone,
-                                             disk=blockdevice_id).execute()
-        except HttpError as e:
-            if e.resp.status == 404:
-                # TODO(mewert) Verify with the rest API this is the only way to
-                # get a 404.
-                raise UnknownVolume(blockdevice_id)
-            else:
-                raise e
-        attached_to = _extract_attached_to(disk)
-        if not attached_to:
-            raise UnattachedVolume(blockdevice_id)
-        return attached_to
+        with start_action(
+                action_type=u"flocker:node:agents:gce:get_attached_to",
+                blockdevice_id=blockdevice_id
+        ) as action:
+            try:
+                # TODO(mewert) verify timeouts and error conditions.
+                disk = self._compute.disks().get(
+                    project=self._project,
+                    zone=self._zone,
+                    disk=blockdevice_id
+                ).execute()
+            except HttpError as e:
+                if e.resp.status == 404:
+                    # TODO(mewert) Verify with the rest API this is the only
+                    # way to get a 404.
+                    raise UnknownVolume(blockdevice_id)
+                else:
+                    raise e
+            attached_to = _extract_attached_to(disk)
+            if not attached_to:
+                raise UnattachedVolume(blockdevice_id)
+            action.add_success_fields(attached_to=attached_to)
+            return attached_to
 
     def detach_volume(self, blockdevice_id):
-        attached_to = self._get_attached_to(blockdevice_id)
-        # TODO(mewert): Test this race (something else detaches right at this
-        # point). Might involve putting all GCE interactions behind a zope
-        # interface and then using a proxy implementation to inject code.
-        self._do_blocking_operation(
-            self._compute.instances().detachDisk, instance=attached_to,
-            deviceName=blockdevice_id)
-        return None
+        with start_action(
+                action_type=u"flocker:node:agents:gce:detach_volume",
+                blockdevice_id=blockdevice_id,
+        ):
+            attached_to = self._get_attached_to(blockdevice_id)
+            # TODO(mewert): Test this race (something else detaches right at
+            # this point). Might involve putting all GCE interactions behind a
+            # zope interface and then using a proxy implementation to inject
+            # code.
+            self._do_blocking_operation(
+                self._compute.instances().detachDisk, instance=attached_to,
+                deviceName=blockdevice_id)
+            return None
 
     def get_device_path(self, blockdevice_id):
         # TODO(mewert): Verify that we need this extra API call.
```

Comment:
+1, decided to keep it as it happens infrequently enough. And... I might as well try to pass the test we have our external partners pass.